olive/passes/pytorch/slicegpt.py

```diff
@@ -0,0 +1,165 @@
+
+    This pass only supports PyTorchModelHandler with hf_config.
+    """
+
+    @staticmethod
+    def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigParam]:
+        return {
+            "calibration_data_config": PassConfigParam(
+                type_=Union[DataConfig, Dict],
+                required=True,
+                description=("Data config for Dataset to calibrate and calculate perplexity on."),
+            ),
+            "calibration_nsamples": PassConfigParam(
+                type_=int,
+                required=False,
+                default_value=128,
+                description=("Number of samples of the calibration data to load."),
+            ),
+            "calibration_batch_size": PassConfigParam(
+                type_=int,
+                required=False,
+                default_value=16,
+                description=("Batch size for loading the calibration data."),
+            ),
+            "calibration_max_seqlen": PassConfigParam(
+                type_=int,
+                required=False,
+                default_value=2048,
+                description=("Maximum sequence length for the calibration data."),
+            ),
+            "varied_seqlen": PassConfigParam(
+                type_=bool,
+                required=False,
+                default_value=False,
+                description=("Varied sequence lengths in the calibration data."),
+            ),
+            "seed": PassConfigParam(
+                type_=int,
+                required=False,
+                default_value=42,
+                description=("Seed for sampling the calibration data."),
+            ),
+            "sparsity": PassConfigParam(
+                type_=float,
+                default_value=0.0,
+                description="A measure of how much slicing is applied (in the range [0, 1))",
+            ),
+            "round_interval": PassConfigParam(
+                type_=int,
+                default_value=8,
+                description="Interval for rounding the weights (the best value may depend on your hardware)",
+            ),
+            "final_orientation": PassConfigParam(
+                type_=str,
+                default_value="random",
+                description="Final orientation of the sliced weights. Choices are random or pca.",
+            ),
+        }
+
+    @torch.no_grad()
+    def _run_for_config(
+        self, model: PyTorchModelHandler, data_root: str, config: Dict[str, Any], output_model_path: str
+    ) -> PyTorchModelHandler:
+        if sys.version_info < (3, 10):
+            raise ValueError("SliceGPT requires python3.10 or higher")
+
+        from slicegpt import layernorm_fusion, rotate
+        from slicegpt.data_utils import get_dataset, prepare_dataloader
+        from slicegpt.hf_utils import get_model_and_tokenizer
+        from slicegpt.slicing_scheduler import ConstSlicingScheduler
+
```

Comment:
Could we move these imports at top since we have lazy loading of passes now in place?

Reply:
I had to revert this change and put the imports back in function local scope.

This pass requires Python >= 3.10 and CI currently uses v3.8 so none of the tests nor the example can be tested on CI. Specific test to check whether or not all the passes can be imported fails with these imports in global/file scope because the requirements file specifically exclude installing SliceGPT for a python version less than the required 3.10. So the tests fails with missing slicegpt import.